lab_1/main.py

```diff
@@ -1,45 +1,44 @@
-def calculate_frequencies(info):
-    words = info.split(' ')
+
+def calculate_frequences(info):
+    wrongs = "1234567890~@#$%^&*\"[]{}\'/\n:;!?().,<>"
     frequencies_dictionary = {}
-    for count, k in enumerate(words):
-        if words[count] in frequencies_dictionary:
-            frequencies_dictionary[words[count]] += 1
-        else:
-            frequencies_dictionary[words[count]] = 1
+    if type(info) is str:
+        text = info.lower()
+        for i in wrongs:
+            text = text.replace(i, "")
+        words = text.split()
+        for count, k in enumerate(words):
+            if words[count] in frequencies_dictionary:
+                frequencies_dictionary[words[count]] += 1
+            else:
+                frequencies_dictionary[words[count]] = 1
     return frequencies_dictionary
 
 
-pass
+def filter_stop_words(freq_dict, stopwords):
+    frequencies = {}
+    if not stopwords:
+        stopwords = []
+    if freq_dict:
+        frequencies = {key: freq_dict[key] for key in freq_dict if (key not in stopwords) and (type(key) is str)}
+    return frequencies
 
 
-def filter_stop_words(frequencies_dict, stopwords):
-    frequencies = {key: frequencies_dict[key] for key in frequencies_dict if key not in stopwords}
-    sorted_words = sorted(frequencies, key=frequencies.get)
-    sorted_words.reverse()
-    return sorted_words
+def get_top_n(sorted_words, top_n):
+    words = sorted(sorted_words.items(), key=lambda item: item[1], reverse=True)
+    if top_n > len(words):
+        top_n = len(words)
+    output = tuple(words[i][0] for i in range(top_n))
+    return output
 
 
-pass
+def read_from_file(path_to_file, lines_limit):
+    text = open(path_to_file)
+    info = text.read(lines_limit)
```

Comment:
@gradevonn I guess you meant 'readlines', otherwise I get just n words, not n lines.